src/context/ProductContext.js

```diff
@@ -0,0 +1,33 @@
+const { createContext, useContext, useEffect, useState } = require("react");
+
+const ProductContext = createContext()
+
+
+export const ProductProvider = ({children}) => {
+      
+    const[ product , setProduct] = useState([])
+    const[category , setCategory] = useState([])
+    useEffect(()=>{
+        fetch ('/api/products')
```

Comment:
You can also use axios for that it is easy.

Reply:
will change it for time being ,  running late for now , have sort of time